### PR TITLE
fix: add Mistral cost tracking

### PR DESCRIPTION
**Pull request by aduverger**
## Details
This PR enables Mistral provider for cost tracking. 

Lots of Mistral's models cost were already provided by LiteLLM, but their 2 flagship models were missing: [Mistral Small 4](https://docs.mistral.ai/models/model-cards/mistral-small-4-0-26-03) (best for speed) and [Mistral Medium 3.5](https://docs.mistral.ai/models/model-cards/mistral-medium-3-5-26-04) (best for performance).

Also, because Mistral was missing from `CostService.java`, the costs were not tracked by SDKs.

## Change checklist
- [N/A] User facing
- [N/A] Documentation update

## AI-WATERMARK

AI-WATERMARK: yes

  - Tools: Claude-Code
  - Model: Opus-4.7
  - Scope: All files
  - Human verification: Yes

## Testing
How you tested

Include:
- Exact commands run (for example: `mvn test`, `npm run lint && npm run test`, `pytest ...`)
- Scenarios validated (happy path, edge cases, regressions)
- Environment/context used (local process mode vs Docker, OS/browser when relevant)
- Any tests not run, with reason
- Relevant logs/artifacts links if available

### CODE REVIEW

File: apps/opik-backend/src/main/resources/model_prices_overrides.json [hunk 1]
```diff
@@ -122,5 +122,26 @@
...
+    "mistral/mistral-medium-3-5": {
+        "input_cost_per_token": 1.5e-06,
+        "litellm_provider": "mistral",
+        "max_input_tokens": 262144,
+        "max_output_tokens": 262144,
+        "max_tokens": 262144,
+        "mode": "chat",
+        "output_cost_per_token": 7.5e-06,
+        "source": "https://mistral.ai/pricing"
+    },
```
[the remark this answers is not in the document]
Comment (aduverger) [reply apps/opik-backend/src/main/resources/model_prices_overrides.json:136]: Verified — the gap is real, but it's not specific to this PR and it's not reachable today for Mistral (`MISTRAL` isn't a `LlmProvider`, so the playground can't issue these requests). 

The actually-affected entry is the pre-existing override for `gemini/gemini-3-flash`.
Filing as a follow-up: extend `ModelCapabilities.loadCapabilities()` to merge `model_prices_overrides.json`, same pattern as `CostService.applyOverrides()`. 
Keeping this PR scoped to cost tracking.

File: apps/opik-backend/src/main/resources/model_prices_overrides.json
Comment (andrescrz) [inline apps/opik-backend/src/main/resources/model_prices_overrides.json:1]: Thanks for the contribution!

Note: `model_prices_overrides.json` is automatically synchronized from the LiteLLM repository, so changes to this file shouldn't be made directly in Opik. Any updates to model pricing should be submitted upstream to LiteLLM instead, and they'll be picked up during the sync process.

Could you remove the changes to this file from the PR?

File: apps/opik-frontend/src/data/model_prices_and_context_window.json
Comment (andrescrz) [inline apps/opik-frontend/src/data/model_prices_and_context_window.json:1]: Applies to this file as well.

**Review by andrescrz — CHANGES_REQUESTED** [review]
Hi @aduverger 

Thanks for your contribution. 

Please see related comments and let me know if any question.

Regards

File: apps/opik-backend/src/main/resources/model_prices_overrides.json
Comment (aduverger) [reply apps/opik-backend/src/main/resources/model_prices_overrides.json:1]: Thanks @andrescrz . I'm a bit confused, my understanding was that the sync workflow `.github/workflows/span_cost_upload_daily.yml` only curls + cps the main pricing file (lines 22–23). It never touches the overrides file. 
From `CostService.java:210` comments: _"Apply Opik-owned overrides that survive the daily LiteLLM sync workflow."_ 
And every existing override entry is also opik-owned: `gpt-oss` isn't upstream in LiteLLM either, yet Opik adds it via this exact file.

File: apps/opik-frontend/src/data/model_prices_and_context_window.json
Comment (aduverger) [reply apps/opik-frontend/src/data/model_prices_and_context_window.json:1]: Good point! Reverted

File: apps/opik-backend/src/main/resources/model_prices_overrides.json
Comment (andrescrz) [reply apps/opik-backend/src/main/resources/model_prices_overrides.json:1]: Sorry, ignore this comment. This doesn't apply for the override file.

**Review by andrescrz — APPROVED** [review]
Hi @aduverger 

Thanks for the contribution! I reviewed the PR and everything looks good from my side. I've approved it, and all CI checks have passed successfully.

We'll merge it shortly, and the change will be included in the next Opik release (v2.0.61), which we're currently expecting to ship on the next 24h approximately.

Apologies for the confusing feedback on the 1st review and thanks again for contributing to Opik!

Best
